### PR TITLE
Added more storage tests

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]

File: component/azstorage/block_blob.go [hunk 1]
```diff
@@ -507,9 +507,10 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 		if !dirList[blobInfo.Name] {
 			//log.Info("BlockBlob::List : meta file does not exists for dir %s", blobInfo.Name)
 			// For these dirs we get only the name and no other properties so hardcoding time to current time
+			name := strings.TrimSuffix(blobInfo.Name, "/")
 			attr := &internal.ObjAttr{
-				Path:  split(bb.Config.prefixPath, blobInfo.Name),
-				Name:  filepath.Base(blobInfo.Name),
+				Path:  split(bb.Config.prefixPath, name),
```
Comment (vibhansa-msft) [inline component/azstorage/block_blob.go:512]: This is a working code, do we have some bug here ?

Comment (gapra-msft) [reply component/azstorage/block_blob.go:512]: I think we expect names to not have a trailing / so I am trimming it. Storage returns these dir prefix names with a / at the end.

Comment (gapra-msft) [reply component/azstorage/block_blob.go:512]: This is to make the test case TestReadDirNoVirtualDirectory get the expected result

File: component/azstorage/azstorage.go [hunk 2]
```diff
@@ -429,6 +430,10 @@ func (az *AzStorage) Chown(options internal.ChownOptions) error {
 
 func (az *AzStorage) FlushFile(options internal.FlushFileOptions) error {
 	log.Trace("AzStorage::FlushFile : Flush file %s", options.Handle.Path)
+	if options.Handle.CacheObj.BlockOffsetList.SmallFile() {
```
Comment (gapra-msft) [inline component/azstorage/azstorage.go:433]: confirmed with Tamer that this is not a valid path to hit

**Review by vibhansa-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

**Pull request merged**